packages/arb-token-bridge-ui/src/hooks/useSigners.ts

```diff
@@ -0,0 +1,68 @@
+import { useMemo } from 'react'
+import { useWallet } from '@gimmixorg/use-wallet'
+import { providers } from 'ethers'
+import { JsonRpcSigner } from '@ethersproject/providers/lib/json-rpc-provider'
+
+import { useNetworks, UseNetworksStatus } from './useNetworks'
+import { rpcURLs } from '../util/networks'
+
+export enum UseSignersStatus {
+  NETWORK_NOT_CONNECTED = 'network_not_connected',
+  NETWORK_NOT_SUPPORTED = 'network_not_supported',
+  SUCCESS = 'success'
+}
+
+export type UseSignersDataUnknown = {
+  l1Signer: undefined
+  l2Signer: undefined
+}
+
+export type UseSignersData = {
+  l1Signer: JsonRpcSigner
+  l2Signer: JsonRpcSigner
+}
+
+export type UseSignersResult =
+  | ({ status: UseSignersStatus.NETWORK_NOT_CONNECTED } & UseSignersDataUnknown)
+  | ({ status: UseSignersStatus.NETWORK_NOT_SUPPORTED } & UseSignersDataUnknown)
+  | ({ status: UseSignersStatus.SUCCESS } & UseSignersData)
+
+const defaults: UseSignersDataUnknown = {
+  l1Signer: undefined,
+  l2Signer: undefined
+}
+
+export function useSigners(): UseSignersResult {
+  const networks = useNetworks()
+  const { account: address, provider } = useWallet()
+
+  return useMemo(() => {
+    if (networks.status === UseNetworksStatus.NOT_CONNECTED) {
+      return { status: UseSignersStatus.NETWORK_NOT_CONNECTED, ...defaults }
+    }
+
+    if (networks.status === UseNetworksStatus.NOT_SUPPORTED) {
+      return { status: UseSignersStatus.NETWORK_NOT_SUPPORTED, ...defaults }
+    }
+
+    const { l1Network, l2Network, isConnectedToArbitrum } = networks
+    const partnerNetwork = isConnectedToArbitrum ? l1Network : l2Network
+    const partnerProvider = new providers.JsonRpcProvider(
+      partnerNetwork.rpcURL || rpcURLs[partnerNetwork.chainID]
```

Comment:
why is the rpcURLs object necessary?

Reply:
In **@arbitrum/sdk**, the rpcUrls are set through environment variables: https://github.com/OffchainLabs/arbitrum-sdk/blob/master/src/lib/dataEntities/networks.ts#L147

But CRA has a limitation - [it only exposes environment variables that are prefixed with **REACT_APP_**, so we can't set those](https://create-react-app.dev/docs/adding-custom-environment-variables/). This seemed like the easiest workaround for now. Later we can adjust the build step to make this work differently.